sql/analyzer/validate_create_table.go

```diff
@@ -146,6 +173,36 @@ func validateModifyColumn(intialSch sql.Schema, schema sql.Schema, mc *plan.Modi
 		return nil, err
 	}
 
+	// TODO: When a column is being modified, we should ideally check that any existing table check constraints
+	//       are still valid (e.g. if the column type changed) and throw an error if they are invalidated.
+	//       That would be consistent with MySQL behavior.
+
+	return newSch, nil
+}
+
+func validateDropColumn(initialSch, sch sql.Schema, dc *plan.DropColumn) (sql.Schema, error) {
+	table := dc.Child
+	nameable := table.(sql.Nameable)
+
+	// Iterate over checks and make sure none of them reference dc.Column
+	for _, check := range dc.Checks {
```

Comment:
Extract a method for this and the other identical block?

Reply:
✅